realtimenet/engine.py

```diff
@@ -67,7 +100,27 @@ def run(self):
                     print("*** Unused predictions ***")
                 self._queue_out.put(predictions, block=False)
 
-    def infer(self, clip):
+    def infer(self, clip: np.ndarray) -> Union[np.ndarray, List[np.ndarray]]:
+        """
+        Infer and return predictions given the input clip from video source.
+        Note that the output is either a numpy.ndarray type or a list consisting
+        of numpy.ndarray.
+
+        For an inference engine running 1 classifier, the output is a numpy.ndarray
+        of shape (row, col), where `col` represents the number of labels that the
+        classifier predicts.
+
+        For an inference engine running more than 1 classifier, the output is a list
+        of N numpy.ndarray, where `N` represents the number of classifiers that the
+        inference engine runs. Each of the numpy.ndarray is of shape (row, col),
+        where `col` represents the number of labels that the classifier predicts.
```

Comment:
I just have a problem with using the words "classifier" / "number of labels that the classifier predicts" because this assumes a classification task. Our MET predictor, for instance, is solving a regression task and doesn't have "labels". 

I think we can rephrase this using more generic terms. How about something like this?

```
For an inference engine running a neural network producing a single output, the returned 
object is a numpy.ndarray of shape (T, C), where `C` represents the number of output 
channels while `T` represents the number of time steps and depends on the length 
of the provided input clip.

For an inference engine running a multi-output neural network, the returned object is a 
list of numpy.ndarray, one for each output.
```

Reply:
Great! I edited it a bit as well :)